.github/workflows/release.yml

```diff
@@ -0,0 +1,45 @@
+name: Release Build
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.23'
+
+      - name: Install dependencies
+        run: go mod download
+
+      - name: Get version
+        id: get_version
+        run: |
+          echo "VERSION=$(date +'v%Y.%m.%d-%H%M')" >> $GITHUB_ENV
```

Comment:
I'd prefer to use semver. I.e. release pipeline may be triggered manually, in this case version will be retrieved from input data.
Other variant is to create release no tag push (only tags of semver format).

Reply:
It’s up to you; I’m only offering my suggestion.
This workflow creates only a draft release, so you can make any changes before publishing.

Reply:
I've updated the workflow logic. It still creates a draft release but now uses semantic versioning for release tags, incrementing the previous tag automatically (though it can still be changed manually). The release name includes the current commit hash, and GitHub's release notes generation is used to populate the release body.